packages/fx-core/src/plugins/resource/bot/appStudio/appStudio.ts

```diff
@@ -200,17 +212,23 @@ export class AppStudio {
   public static async updateMessageEndpoint(
     accessToken: string,
     botId: string,
-    registration: IBotRegistration
+    endpoint: string
   ): Promise<void> {
     const axiosInstance = AppStudio.newAxiosInstance(accessToken);
 
+    const botReg = await AppStudio.getBotRegistration(accessToken, botId);
+    if (!botReg) {
+      throw new MessageEndpointUpdatingError(endpoint);
+    }
+    botReg.messagingEndpoint = endpoint;
+
     let response = undefined;
     try {
       response = await RetryHandler.Retry(() =>
-        axiosInstance.post(`${AppStudio.baseUrl}/api/botframework/${botId}`, registration)
+        axiosInstance.post(`${AppStudio.baseUrl}/api/botframework/${botId}`, botReg)
```

Comment:
Does app studio have some api to only update/patch for endpoint field?

Reply:
As far as I know, the answer is no.
1. Updating message endpoint on Teams Dev Portal also used the same REST API as it is here.
2. It seems no other API except this one to update bot registration configs.

Will double confirm with Ning and look into AppStudio's code later on, and let's assume the answer is no for now.